pythonx/jedi_vim.py

```diff
@@ -340,24 +336,65 @@ def goto(mode="goto", no_output=False):
                                     repr(PythonToVimStr(old_wildignore)))
             vim.current.window.cursor = d.line, d.column
     else:
-        # multiple solutions
-        lst = []
-        for d in definitions:
-            if d.in_builtin_module():
-                lst.append(dict(text=PythonToVimStr('Builtin ' + d.description)))
-            elif d.module_path is None:
-                # Typically a namespace, in the future maybe other things as
-                # well.
-                lst.append(dict(text=PythonToVimStr(d.description)))
-            else:
-                lst.append(dict(filename=PythonToVimStr(d.module_path),
-                                lnum=d.line, col=d.column + 1,
-                                text=PythonToVimStr(d.description)))
-        vim_eval('setqflist(%s)' % repr(lst))
-        vim_eval('jedi#add_goto_window(' + str(len(lst)) + ')')
+        show_goto_multi_results(definitions)
     return definitions
 
 
+def relpath(path):
+    """Make path relative to cwd if it is below."""
+    abspath = os.path.abspath(path)
+    if abspath.startswith(os.getcwd()):
+        return os.path.relpath(path)
+    return path
```

Comment:
I like the relpath change a lot.

In the future we could probably start being even more clever and return a relative path if it's only one/two levels outside or something like that, but I'm not even sure that's a good idea, so thumbs up :)

I actually thought about something similar as well, I would probably use relpath everywhere where we use `escape_file_path` (or even in that function), because I generally want VIM to use short paths for displaying a buffer/tab path. Is there some issue that stops us from doing that?

Reply:
I've thought there would be a better function for it already, and/or that the function itself could be better - just came up very quickly with it.
In Vimscript there would be `fnamemodify(…, ':p:.')` - and `":~:."` to have it relative to HOME.

If you like it like it is, then I think we're good.

I somehow dislike that the window closes automatically when leaving it - I often would like to keep it open for later, but it is easy to comment out for me, and also to retrigger it.
However, maybe something to consider?

Reply:
I would probably leave it for now as it is and wait for feedback. We could also use other keys like `o` (buffer), `t` (tab) to open the buffer at that point and keep the window open.